CRM/Member/Form/Membership.php

```diff
@@ -1940,4 +1928,19 @@ protected function setStatusMessage($membership, $endDate, $receiptSend, $member
     }
   }
 
+  /**
+   * @return bool
+   */
+  protected function isUpdateToExistingRecurringMembership() {
+    $isRecur = FALSE;
+    if ($this->_action & CRM_Core_Action::UPDATE
+      && CRM_Core_DAO::getFieldValue('CRM_Member_DAO_Membership', $this->_id,
+        'contribution_recur_id')
+      && !CRM_Member_BAO_Membership::isSubscriptionCancelled($this->_id)) {
```

Comment:
Is this function coming out of some existing code?  I *think* it makes sense but I need to be sure it's not a change in behaviour? - ie. this is assuming that a cancelled membership cannot be renewed and that may not previously have been the case (even if it's the right thing to do)

Reply:
@mattwire check old lines 666 - 671